integration/dynamic_namespace_test.go

```diff
@@ -0,0 +1,148 @@
+// +build integration
+
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/m3db/m3cluster/integration/etcd"
+	"github.com/m3db/m3db/integration/generate"
+	"github.com/m3db/m3db/storage/namespace"
+	"github.com/m3db/m3db/storage/namespace/convert"
+	"github.com/m3db/m3db/storage/namespace/dynamic"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDynamicNamespaceAdd(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow() // Just skip if we're doing a short run
+	}
+
+	// test options
+	tickInterval := time.Second
+	testOpts := newTestOptions(t).SetTickInterval(tickInterval)
+	require.True(t, len(testOpts.Namespaces()) >= 2)
+	ns0 := testOpts.Namespaces()[0]
+	ns1 := testOpts.Namespaces()[1]
+
+	// embedded kv
+	embeddedKV, err := etcd.New(etcd.NewOptions())
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, embeddedKV.Close())
+	}()
+	require.NoError(t, embeddedKV.Start())
+	csClient, err := embeddedKV.ConfigServiceClient()
+	require.NoError(t, err)
+	kvStore, err := csClient.KV()
+	require.NoError(t, err)
+
+	// namespace maps
+	protoKey := func(nses ...namespace.Metadata) proto.Message {
+		nsMap, err := namespace.NewMap(nses)
+		require.NoError(t, err)
+		p := convert.ToProto(nsMap)
+		return &p
```

Comment:
Perhaps `ToProto` should return a pointer type?  All the `proto` APIs take pointer types so it'll likely always be an awkward second var cast to use anything returned from `ToProto`.